### ISSUE TITLE
Compatible versions for each component

### ISSUE THREAD

**Issue by mplover** [dapr/docs#3355]
**What content needs to be created or modified?**
A table or listing of compatible versions of the component services

**Describe the solution you'd like**
A table with references to the DAPR runtime version and the min/max versions of the related services that are compatible or not compatible (ie. Redis Cache username supported in 6 and above, and Redis streams not supported before 5.0)

**Where should the new material be placed?**
At the bottom of the page. It's not super pertinent for every case, but it would be good to know which versions of the relevant software is compatible for upgrading/downgrading.

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**
N/A

**Additional context**
This came about with my company looking to upgrade Redis from version 5 to 6. There's an assumption that it should be compatible, but I don't know that for certain without digging into specific tabs on the page to know what is and isn't compatible at certain versions. The Redis state store page doesn't give me a minimum version unless I look at the Kubernetes tab where it says to install a minimum version of 5, but there's no mention of max version or if the latest versions are or are not compatible. If it were a table on the bottom of the page, it would be a link in the page navigation to jump or link directly to and be easier to present to infrastructure/architecture.

**Comment by msfussell** [comment]
@mplover - I see. So you are asking the SDK version supported in each of the components. Seems fair, but is hard to update for each component and this only applies to certain components, for example public cloud services always remain current. Did you have another example other than Redis?

**Comment by msfussell** [comment]
Revisited this and yes, this is important information to capture. A similar issue was brought up for the Apache Kafka component (which version are supported). We should consider adding a table of support versions for most components starting with Redis and Kafka.

**Comment by mplover** [comment]
Exactly. I assume this would just be a task of going through each sdk used for each component across the various versions of dapr and interpreting that information into a table.
